packages/notification-services-controller/src/NotificationServicesController/NotificationServicesController.ts

```diff
@@ -1096,35 +1096,46 @@ export default class NotificationServicesController extends BaseController<
     try {
       this.#setIsFetchingMetamaskNotifications(true);
 
+      // This is used by Feature Announcement & On Chain
+      // Not used by Snaps
+      const isGlobalNotifsEnabled = this.state.isNotificationServicesEnabled;
+
       // Raw Feature Notifications
-      const rawFeatureAnnouncementNotifications = this.state
-        .isFeatureAnnouncementsEnabled
-        ? await FeatureNotifications.getFeatureAnnouncementNotifications(
-            this.#featureAnnouncementEnv,
-            previewToken,
-          ).catch(() => [])
-        : [];
+      const rawFeatureAnnouncementNotifications =
+        isGlobalNotifsEnabled && this.state.isFeatureAnnouncementsEnabled
```

Comment:
Added `isGlobalNotifsEnabled` check to make sure we fetch announcements if the global flag is enabled.